wasm_filter/src/config.rs

```diff
@@ -0,0 +1,90 @@
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct MappingRule {
+    pattern: std::string::String,
+    http_method: std::string::String, // @TODO this should be a enum, maybe from hyper
+    metric_system_name: std::string::String,
+    delta: u32,
+}
+
+impl MappingRule {
+    fn matches(&self, method: std::string::String, path: std::string::String) -> bool {
+        log::debug!(
+            "MappingRule:Match: METHOD:: '{}', PATH:: '{}', MappingRULE: '{:?}'",
+            method,
+            path,
+            self
+        );
+
+        if self.http_method != method {
+            return false;
+        }
+        // @TODO should be a regexp
+        if self.pattern != path {
+            return false;
+        }
+        true
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Default)]
+pub struct Service {
+    pub id: u32,
+    pub hosts: Vec<std::string::String>,
+    pub policies: Vec<std::string::String>,
+    pub target_domain: std::string::String,
+    pub proxy_rules: Vec<MappingRule>,
+}
+
+impl Service {
+    pub fn match_mapping_rule(
+        &self,
+        method: std::string::String,
+        path: std::string::String,
+    ) -> (bool, std::string::String) {
+        let mut metrics: HashMap<std::string::String, u32> = HashMap::new();
+        for mapping_rule in &self.proxy_rules {
+            if mapping_rule.matches(method.clone(), path.clone()) {
+                log::debug!("Mapping rule matches: {:?}", mapping_rule);
+                metrics.insert(mapping_rule.metric_system_name.clone(), mapping_rule.delta);
+            }
+        }
+        if metrics.len() > 0 {
+            return (true, serde_json::to_string(&metrics).unwrap());
+        }
+        (false, serde_json::to_string(&metrics).unwrap())
+    }
+}
+
+thread_local! {
```

Comment:
We should document why we need a `thread_local` here. Also perhaps we can move the configuration to `ConfigContext`.

Reply:
It's a global var and ConfigContext cannot be shared on HTTPContext.